inject/android.js

```diff
@@ -16,10 +25,43 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    const messageSecret = processedConfig.messageSecret
+    const messageCallback = processedConfig.messageCallback
+    const messageInterface = processedConfig.messageInterface
+
+    const wrappedUpdate = ((providedSecret, ...args) => {
+        if (providedSecret === messageSecret) {
+            update(...args)
+        }
+    }).bind()
+
+    Object.defineProperty(window, messageCallback, {
+        value: wrappedUpdate
+    })
+
     init(processedConfig)
 
-    // Not supported:
-    // update(message)
+    window.addEventListener('sendMessageProxy' + messageSecret, event => {
+        event.stopImmediatePropagation()
+
+        if (!(event instanceof CustomEvent) || !event?.detail) {
+            return console.warn('no details in sendMessage proxy', event)
+        }
+
+        const messageType = event.detail?.messageType
+        if (!allowedMessages.includes(messageType)) {
+            return console.warn('Ignoring invalid sendMessage messageType', messageType)
+        }
+
+        const message = {
+            type: messageType,
+            options: event.detail?.options,
+            secret: messageSecret
+        }
+        const stringifiedArgs = JSON.stringify(message)
+        // @ts-ignore
+        window[messageInterface].process(stringifiedArgs)
```

Comment:
```suggestion
        // Earlier in the file...
        const sendMessageToAndroid = window[messageInterface].process.bind(window[messageInterface])
        
        // Here
        sendMessageToAndroid(secret, stringifiedArgs)
```

That's about as good as we can do to prevent them messing with `window.messageInterface` + has the bonus of giving the wrapped function a clearer name.